restapi/src/main/java/io/stargate/web/resources/AuthenticatedDB.java

```diff
@@ -78,4 +79,38 @@ public AuthenticationSubject getAuthenticationSubject() {
   public void setAuthenticationSubject(AuthenticationSubject authenticationSubject) {
     this.authenticationSubject = authenticationSubject;
   }
+
+  /**
+   * Retrieve user defined types definitions for a keyspace.
+   *
+   * @param keyspaceName existing keyspace name
+   * @return a collection of user defined types definitions
+   */
+  public Collection<UserDefinedType> getTypes(String keyspaceName) {
+    Keyspace keyspace = getKeyspace(keyspaceName);
+    if (keyspace == null) {
+      throw new NotFoundException(String.format("keyspace '%s' not found", keyspaceName));
+    }
+    return keyspace.userDefinedTypes();
+  }
+
+  /**
+   * Retrieve user defined types definitions from its identifier in a keyspace.
+   *
+   * @param keyspaceName existing keyspace name
+   * @param typeName identifier for the type
+   * @return a collection of user defined types definitions
+   */
+  public UserDefinedType getType(String keyspaceName, String typeName) {
+    Keyspace keyspace = getKeyspace(keyspaceName);
+    if (keyspace == null) {
+      throw new NotFoundException(String.format("keyspace '%s' not found", keyspaceName));
+    }
+
+    UserDefinedType typeMetadata = keyspace.userDefinedType(typeName);
+    if (typeMetadata == null) {
+      throw new NotFoundException(String.format("table '%s' not found", typeName));
```

Comment:
#nitpick Will this be better if we modify it to be `...String.format("type '%s' not found in the keyspace '%s'", typeName, keyspaceName)...`?

Reply:
Gonna change that. Thanks! 🙂